### DISCUSSION TITLE
How to write content of an url html page to one pdf page directly? Without parsing html code? Just like put an iframe, etc.?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#4153] [Q&A]
How to write content of an url html page to one pdf page directly? Without parsing html code? Just like put an iframe, etc.?

**Comment by JorjMcKie** [comment]
As HTML source or should it be rendering like the web page or what.

**Comment by nissansz** [comment]
Like a printed clean web page. Insert a link to render it as a clean pdf page.

**Reply by JorjMcKie** [reply]
ok. No there is no way to achieve this.

**Comment by nissansz** [comment]
Can copy whole page html source, and put in a div to write html and show in pdf?

**Reply by JorjMcKie** [reply]
You could write a script that uses `Story` for this.
But the remaining problem here would be its support for only a subset of HTML4 / CSS2.
You also can open a HTML file as a PyMuPDF document, then use `doc.convert_to_pdf()`.
This will however have similar and many other issues.

**Comment by nissansz** [comment]
Hope one day, it is possible to use html/canvas directly in pymupdf without issues.
